### PR TITLE
dac: ad5791: Add APIs for clear, LDAC

### PR DESCRIPTION
**Pull request by SaikiranGudla**
Add ldac toggle and clear async apis and fix
set_lin_comp api

## Pull Request Description

Please replace this with a detailed description and motivation of the changes. 
You can tick the checkboxes below with an 'x' between square brackets or just check them after publishing the PR. 
If this PR contains a breaking change, list dependent PRs and try to push all related PRs at the same time.

## PR Type
- [x] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: drivers/dac/ad5791/ad5791.c [hunk 1]
```diff
@@ -408,11 +408,67 @@ int ad5791_set_lin_comp(struct ad5791_dev *dev,
 		return -EINVAL;
 	}
 
-	if (!v_span)
+	if (v_span)
```
Comment (rbolboac) [inline drivers/dac/ad5791/ad5791.c:411]: this fix could be in a different commit, with the according fixes tag

File: drivers/dac/ad5791/ad5791.c [hunk 2]
```diff
@@ -408,11 +408,67 @@ int ad5791_set_lin_comp(struct ad5791_dev *dev,
...
+	}
+
+	/* If no gpio is assigned use SW LDAC */
+	return ad5791_soft_instruction(dev, AD5791_SOFT_CTRL_LDAC);
+}
+
+/***************************************************************************//**
+ * @brief	Clear DAC channel output with the clearcode.
+ *
+ * @param	dev		- The device structure.
```
Comment (rbolboac) [inline drivers/dac/ad5791/ad5791.c:451]: extra blank spaces between dev and description are not needed

File: drivers/dac/ad5791/ad5791.c [hunk 3]
```diff
@@ -408,11 +408,67 @@ int ad5791_set_lin_comp(struct ad5791_dev *dev,
...
 	return ad5791_spi_write_mask(dev,
 				     AD5791_REG_CTRL,
 				     AD5791_CTRL_LINCOMP_MASK,
 				     AD5791_CTRL_LINCOMP(v_span));
 }
+
+/***************************************************************************//**
+ * @brief	Trigger LDAC
+ *
+ * @param	dev		- The device structure.
```
Comment (rbolboac) [inline drivers/dac/ad5791/ad5791.c:423]: extra blank spaces between dev and description are not needed

**Review by rbolboac — COMMENTED** [review]
all in all looks good... some small remarks from my side

File: drivers/dac/ad5791/ad5791.c (hunk 1 above)
Comment (rbolboac) [reply drivers/dac/ad5791/ad5791.c:411]: Please add the fixes tag, above the Signed-off-by tag in the commit message as follows:
```
Fixes: b527970 ("drivers: dac: ad5791: Add linearity error compensation API")
```

**Comment by SaikiranGudla** [comment]
v1: Addressed review comments

1. Removed extra spaces between dev and description
2. Added 2 separate commits
3. Added fix tag

**Review by rbolboac — APPROVED** [review]

**Comment by SaikiranGudla** [comment]
Is this PR good to be merged?
